diff --git a/ports/zlib/portfile.cmake b/ports/zlib/portfile.cmake
--- a/ports/zlib/portfile.cmake
+++ b/ports/zlib/portfile.cmake
@@ -3,7 +3,7 @@ include(vcpkg_common_functions)
 set(VERSION 1.2.11)
 
 vcpkg_download_distfile(ARCHIVE_FILE
-    URLS "http://www.zlib.net/zlib-${VERSION}.tar.gz" "https://downloads.sourceforge.net/project/libpng/zlib/${VERSION}/zlib-${VERSION}.tar.gz"
+    URLS "https://github.com/madler/zlib/archive/refs/tags/v${VERSION}.tar.gz"
     FILENAME "zlib1211.tar.gz"
-    SHA512 73fd3fff4adeccd4894084c15ddac89890cd10ef105dd5e1835e1e9bbb6a49ff229713bd197d203edfa17c2727700fce65a2a235f07568212d820dca88b528ae
+    SHA512 104c62ed1228b5f1199bc037081861576900eb0697a226cafa62a35c4c890b5cb46622e399f9aad82ee5dfb475bae26ae75e2bd6da3d261361b1c8b996970faf
 )
