diff --git a/ports/angle/CMakeLists.txt b/ports/angle/CMakeLists.txt
index 8be51c0d4..11fe02704 100644
--- a/ports/angle/CMakeLists.txt
+++ b/ports/angle/CMakeLists.txt
@@ -370,7 +370,9 @@ if(NOT DISABLE_INSTALL_HEADERS)
     install(
         DIRECTORY include/
         DESTINATION include
-        FILES_MATCHING PATTERN "*.h"
+        FILES_MATCHING
+		PATTERN "*.h"
+		PATTERN "*.inc"
         PATTERN "GLSLANG" EXCLUDE
         PATTERN "EGL" EXCLUDE
         PATTERN "KHR" EXCLUDE
