diff --git a/scripts/cmake/vcpkg_configure_meson.cmake b/scripts/cmake/vcpkg_configure_meson.cmake
index 844bd1eb5..8318e8762 100644
--- a/scripts/cmake/vcpkg_configure_meson.cmake
+++ b/scripts/cmake/vcpkg_configure_meson.cmake
@@ -32,12 +32,19 @@ function(vcpkg_configure_meson)
         set(ENV{PROCESSOR_ARCHITEW6432} "AMD64")
         set(ENV{PROCESSOR_ARCHITECTURE} "AMD64")
         set(_VCPKG_MESON_NATIVE_FILE "${CURRENT_BUILDTREES_DIR}/vcpkg-x64-native.ini")
         file(WRITE "${_VCPKG_MESON_NATIVE_FILE}"
+            "[binaries]\n"
+            "c = 'cl'\n"
+            "cpp = 'cl'\n"
+            "ar = 'lib'\n"
+            "[properties]\n"
+            "needs_exe_wrapper = false\n"
             "[host_machine]\n"
             "system = 'windows'\n"
             "cpu_family = 'x86_64'\n"
             "cpu = 'x86_64'\n"
             "endian = 'little'\n")
         list(APPEND _vcm_OPTIONS --native-file ${_VCPKG_MESON_NATIVE_FILE})
+        set(MESON_COMMON_LDFLAGS "${MESON_COMMON_LDFLAGS} /MACHINE:X64")
     endif()
     if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
